### PR TITLE
Destination e2e test: enable dedup

### PR DESCRIPTION
**Pull request by edgao**
tag destination-e2e as supporting dedup, so that we can use it for more testing things

### CODE REVIEW

**Review by evantahler — APPROVED** [review]
Fake connector faking deduplication? Love to see it.

File: airbyte-integrations/connectors/destination-e2e-test/src/main/resources/spec.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "supportsIncremental": true,
   "supportsNormalization": false,
   "supportsDBT": false,
-  "supported_destination_sync_modes": ["overwrite", "append"],
+  "supported_destination_sync_modes": ["overwrite", "append", "append_dedup"],
```
Comment (teallarson) [inline airbyte-integrations/connectors/destination-e2e-test/src/main/resources/spec.json:6]: 💥

**Review by teallarson — APPROVED** [review]
WOOHOO!
@timroes we'll be able to test both refresh modes in the UI now 😄

**Comment by edgao** [comment]
.... how are tests failing, this connector does literally nothing

**Pull request merged**